buildlib/tools/test_mad.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -exE -o pipefail
+
+IMAGE="rdmz-harbor.rdmz.labs.mlnx/ucx/x86_64/rhel8.2/builder:mofed-5.0-1.0.0.0"
+cd "$BUILD_SOURCESDIRECTORY"
+
+node_setup() {
+    funcname
+    sudo chmod 777 /dev/infiniband/umad*
+}
+
+build_ucx() {
+    funcname
+    ./autogen.sh
+    ./contrib/configure-release \
+        --prefix="$PWD"/install \
+        --with-mad \
+        --without-valgrind \
+        --without-go \
+        --without-java
+    make -s -j"$(nproc)"
+    make install
+}
+
+build_ucx_in_docker() {
+    docker run --rm \
+        --name ucx_build_"$BUILD_BUILDID" \
+        -v "$PWD":"$PWD" -w "$PWD" \
+        $IMAGE \
+        bash -c "source ./buildlib/tools/test_mad.sh && build_ucx"
+
+    sudo chown -R swx-azure-svc:ecryptfs "$PWD"
+}
+
+docker_run_srv() {
+    detect_hca
+    docker run --rm \
+        --detach \
+        --net=host \
+        -e HCA="$HCA" \
+        --name ucx_perftest_"$BUILD_BUILDID" \
+        -v "$PWD":"$PWD" -w "$PWD" \
+        -v /hpc/local:/hpc/local \
+        --gpus all --ulimit memlock=-1:-1 --device=/dev/infiniband/ \
+        $IMAGE \
+        bash -c "source ./buildlib/tools/test_mad.sh && \
+                ${PWD}/install/bin/ucx_perftest -K ${HCA}"
+}
+
+docker_stop_srv() {
+    docker stop ucx_perftest_"$BUILD_BUILDID" || true
```

Comment:
stop is not enough we should remove the container as well, we dont want to have dangling containers all the time

Reply:
We have a daily cleanup of Docker resources.

Reply:
So we will have many stopped containers every day until they are cleaned? this is bad practice, each run should do its best to clean after it self, the daily cleanup is extra cleanup not instead

Reply:
Check out the `--rm ` flag.

Reply:
so why do you stop it? if we need to clean than we need to clean we stop is not cleanning

Reply:
Containers get cleaned once stopped.